File: LeetCode/validParenthesis.py
def isValid(self, s: str) -> bool:
    a=[]
    d={ ')':'(' , ']':'[' , '}':'{' }
    b='([{'
    for i in s:
        if i in b:
            a.append(i)
        else:
            if a and a[-1]==d[i]:
                a.pop()
            else:
                return False
    if a:
        return False
    return True

File: Javascript/Stacks/validParenthesis.js
//https://leetcode.com/problems/valid-parentheses/submissions/

var isValid = function (s) {
    let dict = { ")": "(", "}": "{", "]": "[" };
    let starts = "({[";
    let arr = [];
    for (let i of s) {
        if (starts.includes(i)) {
            arr.push(i);
        } else {
            if (arr.length != 0 && arr[arr.length - 1] == dict[i]) {
                arr.pop();
            } else {
                return false;
            }
        }
    }
    return arr.length == 0;
};

// Input: s = "()[]{}"
// Output: true
